backend/core/src/shared/units-transformations.ts

```diff
@@ -115,92 +118,138 @@ const hmiData = (units: Units, byUnitType: UnitSummary[], amiPercentages: string
     },
     rentAsPercentIncomeRange: {
-      min: parseFloat(unit.monthlyRentAsPercentOfIncome),
-      max: parseFloat(unit.monthlyRentAsPercentOfIncome),
-    },
-    floorRange: {
-      min: unit.floor,
-      max: unit.floor,
+      min: parseFloat(item.monthlyRentAsPercentOfIncome),
+      max: parseFloat(item.monthlyRentAsPercentOfIncome),
     },
-    unitType: unit.unitType,
-    totalAvailable: 0,
-    totalCount: 0,
+    floorRange: floorRange,
+    unitType: item.unitType,
+    totalAvailable: item?.totalAvailable ? item.totalAvailable : 0,
+    totalCount: item?.totalCount ? item.totalCount : 0,
   } as UnitSummary
 }
 
-const getUnitsSummary = (unit: Unit, existingSummary?: UnitSummary) => {
+function getUnitsSummary<T extends BaseUnitInfo>(item: T, existingSummary?: UnitSummary) {
   if (!existingSummary) {
-    return getDefaultSummaryRanges(unit)
+    return getDefaultSummaryRanges(item)
   }
+
   const summary = existingSummary
 
   // Income Range
   summary.minIncomeRange = minMaxCurrency(
     summary.minIncomeRange,
-    getRoundedNumber(unit.monthlyIncomeMin)
+    getRoundedNumber(item.monthlyIncomeMin)
   )
 
   // Occupancy Range
-  summary.occupancyRange = minMax(summary.occupancyRange, unit.minOccupancy)
-  summary.occupancyRange = minMax(summary.occupancyRange, unit.maxOccupancy)
+  summary.occupancyRange = minMax(summary.occupancyRange, item.minOccupancy)
+  summary.occupancyRange = minMax(summary.occupancyRange, item.maxOccupancy)
 
   // Rent Ranges
   summary.rentAsPercentIncomeRange = minMax(
     summary.rentAsPercentIncomeRange,
-    parseFloat(unit.monthlyRentAsPercentOfIncome)
+    parseFloat(item.monthlyRentAsPercentOfIncome)
   )
-  summary.rentRange = minMaxCurrency(summary.rentRange, getRoundedNumber(unit.monthlyRent))
+  summary.rentRange = minMaxCurrency(summary.rentRange, getRoundedNumber(item.monthlyRent))
 
-  // Floor Range
-  if (unit.floor) {
-    summary.floorRange = minMax(summary.floorRange, unit.floor)
+  if (item instanceof Unit) {
+    // Floor Range
+    summary.floorRange = minMax(summary.floorRange, item.floor)
+    // Area Range
+    summary.areaRange = minMax(summary.areaRange, parseFloat(item.sqFeet))
+  } else {
+    // Floor Range
+    summary.floorRange = minMax(summary.floorRange, item.floorMin)
+    summary.floorRange = minMax(summary.floorRange, item.floorMax)
+    // Area Range
+    summary.areaRange = minMax(summary.areaRange, parseFloat(item.sqFeetMin))
+    summary.areaRange = minMax(summary.areaRange, parseFloat(item.sqFeetMax))
+    // Unit Counts
+    summary.totalAvailable += item.totalAvailable ? item.totalAvailable : 0
+    summary.totalCount += item.totalCount ? item.totalCount : 0
   }
 
-  // Area Range
-  summary.areaRange = minMax(summary.areaRange, parseFloat(unit.sqFeet))
-
   return summary
 }
 
-type UnitMap = {
-  [key: string]: Unit[]
+// Shared properties between Unit and UnitsSummary.
+interface BaseUnitInfo {
```

Comment:
It's ok that Unit and UnitsSummary don't actually extend this interface? Is this just a way to sort of fake out Typescript?